serving/using-external-dns.md

```diff
@@ -0,0 +1,119 @@
+[ExternalDNS documentation](https://github.com/kubernetes-incubator/external-dns#deploying-to-a-cluster).
+
+### Choose a DNS provider
+
+Skip this step if you already have a DNS provider for your domain.
+
+Here is a [list](https://github.com/kubernetes-incubator/external-dns#the-latest-release-v05)
+of DNS providers supported by ExternalDNS. Choose a DNS provider from the list.
+
+### Create a DNS zone for managing DNS records
+
+Skip this step if you already have a zone for managing the DNS records of your 
+custom domain.
+
+A DNS zone which will contain the managed DNS records needs to be created.
+Assume your custom domain is `external-dns-test.my-org.do`.
+
+Use the following command to create a DNS zone with [Google Cloud DNS](https://cloud.google.com/dns/):
+```shell
+gcloud dns managed-zones create "external-dns-zone" \
+    --dns-name "external-dns-test.my-org.do." \
+    --description "Automatically managed zone by kubernetes.io/external-dns"
+```
+Make a note of the nameservers that were assigned to your new zone.
+```shell
+gcloud dns record-sets list \
+    --zone "external-dns-zone" \
+    --name "external-dns-test.my-org.do." \
+    --type NS
+```
+You should see output similar to the following:
+```
+NAME                             TYPE  TTL    DATA
+external-dns-test.my-org.do.  NS    21600  ns-cloud-e1.googledomains.com.,ns-cloud-e2.googledomains.com.,ns-cloud-e3.googledomains.com.,ns-cloud-e4.googledomains.com.
+```
+In this case, the DNS nameservers are `ns-cloud-{e1-e4}.googledomains.com`. 
+Yours could differ slightly, e.g. {a1-a4}, {b1-b4} etc.
+
+If this zone has the parent zone, you need to add NS records of this zone into 
+the parent zone so that this zone can be found from the parent.
+Assuming the parent zone is `my-org-do` and the parent domain is `my-org.do`, 
+and the parent zone is also hosted at Google Cloud DNS, you can follow these 
+steps to add the NS records of this zone into the parent zone: 
+```shell
+gcloud dns record-sets transaction start --zone "my-org-do"
+gcloud dns record-sets transaction add ns-cloud-e{1..4}.googledomains.com. \
+    --name "external-dns-test.my-org.do." --ttl 300 --type NS --zone "my-org-do"
+gcloud dns record-sets transaction execute --zone "my-org-do"
+```
+
+### Deploy ExternalDNS
+
+#### Google Cloud DNS
+Apply the [manifest](https://github.com/kubernetes-incubator/external-dns/blob/master/docs/tutorials/gke.md#manifest-for-clusters-without-rbac-enabled) to install ExternalDNS. Note that you need
+to set the argument `domain-filter` to your custom domain.
+
+You should see ExternalDNS is installed by running:
+```shell
+kubectl get deployment external-dns
+```
+
+### Configuring Knative Gateway service
+
+In order to publish the Knative Gateway service, the annotation
+`external-dns.alpha.kubernetes.io/hostname: '*.external-dns-test.my-org.do'`
+needs to be added into Knative gateway service:
```

Comment:
I would show where this is added; you can include an example final version of the file after this addition has been made.

Reply:
Done